tests/e2e_test/status_propagation_test.go

```diff
@@ -1,84 +1,74 @@
 package e2e_test
 
 import (
-	"k8s.io/apimachinery/pkg/api/meta"
+	"context"
 
 	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
-	"github.com/kyma-project/lifecycle-manager/pkg/testutils"
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("Warning Status Propagation", Ordered, func() {
-	kyma := testutils.NewKymaWithSyncLabel("kyma-sample", "kcp-system", "regular",
+	kyma := NewKymaWithSyncLabel("kyma-sample", "kcp-system", "regular",
 		v1beta2.SyncStrategyLocalSecret)
-	GinkgoWriter.Printf("kyma before create %v\n", kyma)
+	moduleName := "template-operator"
+	moduleCR := NewTestModuleCR(remoteNamespace)
 
-	BeforeAll(func() {
-		// make sure we can list Kymas to ensure CRDs have been installed
-		err := controlPlaneClient.List(ctx, &v1beta2.KymaList{})
-		Expect(meta.IsNoMatchError(err)).To(BeFalse())
-	})
+	InitEmptyKymaBeforeAll(kyma)
+	CleanupKymaAfterAll(kyma)
 
-	It("Should create empty Kyma CR on remote cluster", func() {
-		Eventually(CreateKymaSecret).
-			WithContext(ctx).
-			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient).
-			Should(Succeed())
-		Eventually(controlPlaneClient.Create).
-			WithContext(ctx).
-			WithArguments(kyma).
-			Should(Succeed())
-		By("verifying kyma is ready")
-		Eventually(CheckKymaIsInState).
-			WithContext(ctx).
-			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
-			Should(Succeed())
-		By("verifying remote kyma is ready")
-		Eventually(CheckRemoteKymaCR).
-			WithContext(ctx).
-			WithArguments(remoteNamespace, []v1beta2.Module{}, runtimeClient, v1beta2.StateReady).
-			Should(Succeed())
-	})
+	Context("Given Template Operator Default CR", func() {
+		It("When enable Template Operator", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(defaultRemoteKymaName, remoteNamespace, moduleName, kyma.Spec.Channel, runtimeClient).
+				Should(Succeed())
+		})
 
-	It("Should enable Template Operator and Kyma should result in Warning status", func() {
-		By("Enabling Template Operator")
-		Eventually(EnableModule).
-			WithContext(ctx).
-			WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", "regular", runtimeClient).
-			Should(Succeed())
-		By("Checking state of kyma")
-		Eventually(CheckKymaIsInState).
-			WithContext(ctx).
-			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateWarning).
-			Should(Succeed())
+		It("Then module CR exist", func() {
+			Eventually(ModuleCRExists).
+				WithContext(ctx).
+				WithArguments(runtimeClient, moduleCR).
+				Should(Succeed())
+		})
+
+		It("Then resource defined in manifest CR", func() {
```

Comment:
I am confused by the description of this case. Should it be: `Then resource [is] defined in manifest CR` or maybe: `Then resource defined in manifest CR [exists]`?

Reply:
changed to `Then resource is defined in manifest CR`